src/CharterManager.sol

```diff
@@ -282,10 +289,10 @@ contract CharterManagerImp {
 
     function onVatFlux(address gemJoin, address from, address to, uint256 wad) external {}
 
-    function quit(bytes32 ilk, address dst) external {
+    function quit(bytes32 ilk, address u, address dst) external allowed(u) {
         require(VatLike(vat).live() == 0, "CharterManager/vat-still-live");
 
-        address urp = getOrCreateProxy(msg.sender);
+        address urp = proxy[u];
```

Comment:
We might want to consider checking that the `urp` is different than `address(0)`, it is not needed but gives a message error if the user doesn't have a proxy instead of blindly failing on `fork`.

Reply:
Basically we are doing the same thing in `move`. Not strictly necessary to check is different `address(0)` but gives a better error message than `vat.move` failing.